RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitCalibrationAlgorithms.dev.cc

```diff
@@ -0,0 +1,194 @@
+#include <cstddef>
+
+// CMSSW imports
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+// Alpaka imports
+//#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+// HGCal imports
+#include "RecoLocalCalo/HGCalRecAlgos/interface/alpaka/HGCalRecHitCalibrationAlgorithms.h"
+#include "DataFormats/HGCalDigi/interface/HGCalRawDataDefinitions.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  //
+  struct HGCalRecHitCalibrationKernel_flagRecHits {
+    ALPAKA_FN_ACC void operator()(Acc1D const& acc,
+                                  HGCalDigiDevice::View digis,
+                                  HGCalRecHitDevice::View recHits,
+                                  HGCalCalibParamDevice::ConstView calibs) const {
+      for (auto idx : uniform_elements(acc, digis.metadata().size())) {
+        auto calib = calibs[idx];
+        bool calibvalid = calib.valid();
+        auto digi = digis[idx];
+        auto digiflags = digi.flags();
+        //recHits[idx].flags() = digiflags;
+        bool isAvailable((digiflags != hgcal::DIGI_FLAG::Invalid) && (digiflags != hgcal::DIGI_FLAG::NotAvailable) &&
+                         calibvalid);
+        bool isToAavailable((digiflags != hgcal::DIGI_FLAG::ZS_ToA) && (digiflags != hgcal::DIGI_FLAG::ZS_ToA_ADCm1));
+        recHits[idx].flags() = (!isAvailable) * hgcalrechit::HGCalRecHitFlags::EnergyInvalid +
+                               (!isToAavailable) * hgcalrechit::HGCalRecHitFlags::TimeInvalid;
+      }
+    }
+  };
+
+  //
+  struct HGCalRecHitCalibrationKernel_adcToCharge {
+    ALPAKA_FN_ACC void operator()(Acc1D const& acc,
+                                  HGCalDigiDevice::View digis,
+                                  HGCalRecHitDevice::View recHits,
+                                  HGCalCalibParamDevice::ConstView calibs) const {
+      auto adc_denoise =
+          [&](uint32_t adc, uint32_t cm, uint32_t adcm1, float adc_ped, float cm_slope, float cm_ped, float bxm1_slope) {
+            float cmf = cm_slope * (0.5 * float(cm) - cm_ped);
```

Comment:
```suggestion
            float cmf = cm_slope * (0.5f * float(cm) - cm_ped);
```
to avoid a conversion to `double` and back to `float`